op-proposer/proposer/driver.go

```diff
@@ -269,18 +269,48 @@ func (l *L2OutputSubmitter) FetchL2OOOutput(ctx context.Context) (*eth.OutputRes
 	return output, true, nil
 }
 
-// FetchDGFOutput gets the next output proposal for the DGF.
+// FetchDGFOutput queries the DGF for the latest game and infers whether it is time to make another proposal
+// If necessary, it gets the next output proposal for the DGF, and returns it along with
+// a boolean for whether the proposal should be submitted at all.
 // The passed context is expected to be a lifecycle context. A network timeout
 // context will be derived from it.
-func (l *L2OutputSubmitter) FetchDGFOutput(ctx context.Context) (*eth.OutputResponse, error) {
+func (l *L2OutputSubmitter) FetchDGFOutput(ctx context.Context) (*eth.OutputResponse, bool, error) {
 	ctx, cancel := context.WithTimeout(ctx, l.Cfg.NetworkTimeout)
 	defer cancel()
 
-	blockNum, err := l.FetchCurrentBlockNumber(ctx)
+	shouldPropose := false
+
+	gameCount, err := l.dgfContract.GameCount(&bind.CallOpts{Context: ctx})
 	if err != nil {
-		return nil, err
+		l.Log.Warn("Could not query DisputeGameFactory.gameCount(), sending a proposal immediately")
+		shouldPropose = true
```

Comment:
Wondering if in this case, we just want to return the error and try again on the next poll. If we cannot query the DGF, something is wrong. We should double check with a Proofs expert.

Reply:
[e9f2658](https://github.com/ethereum-optimism/optimism/pull/11417/commits/e9f2658eb8eb1c00a463dd3f6f2e98eec54bba83)